utils/graphql/update.go

```diff
@@ -0,0 +1,80 @@
+package graphql
+
+import (
+	"context"
+	"strings"
+
+	"github.com/graphql-go/graphql/language/ast"
+
+	"github.com/spaceuptech/space-cloud/model"
+	"github.com/spaceuptech/space-cloud/utils"
+)
+
+func (graph *Module) execUpdateRequest(field *ast.Field, store m) (m, error) {
+	dbType := field.Directives[0].Name.Value
+	col := strings.TrimPrefix(field.Name.Value, "update_")
+
+	req, err := generateUpdateRequest(field, store)
+	if err != nil {
+		return nil, err
+	}
+	status, err := graph.auth.IsUpdateOpAuthorised(graph.project, dbType, col, "", req)
+	if err != nil {
+		return nil, err
+	}
+
+	return m{"status": status}, graph.crud.Update(context.TODO(), dbType, graph.project, col, req)
+}
+
+func generateUpdateRequest(field *ast.Field, store m) (*model.UpdateRequest, error) {
+	var err error
+	var updateRequest model.UpdateRequest
+
+	//---	 upsert identification is wrong
+	if strings.HasPrefix(field.Name.Value, "upsert") {
```

Comment:
This if else condition isn't needed! Instead make a `extractUpdateOperstion` function which returns the operation. The operation is available as the `op` field in the arguments.